kapitan/cli.py

```diff
@@ -38,250 +39,268 @@
-    lint_parser.add_argument('--fail-on-warning',
-                             default=from_dot_kapitan('lint', 'fail-on-warning', False),
-                             action='store_true',
-                             help='exit with failure code if warnings exist, default is False')
-    lint_parser.add_argument('--skip-class-checks',
-                             action='store_true',
-                             help='skip checking for unused classes, default is False',
-                             default=from_dot_kapitan('lint', 'skip-class-checks', False))
-    lint_parser.add_argument('--skip-yamllint',
-                             action='store_true',
-                             help='skip running yamllint on inventory, default is False',
-                             default=from_dot_kapitan('lint', 'skip-yamllint', False))
-    lint_parser.add_argument('--search-secrets',
-                             default=from_dot_kapitan('lint', 'search-secrets', False),
-                             action='store_true',
-                             help='searches for plaintext secrets in inventory, default is False')
-    lint_parser.add_argument('--secrets-path',
-                             help='set secrets path, default is "./secrets"',
-                             default=from_dot_kapitan('lint', 'secrets-path', './secrets'))
-    lint_parser.add_argument('--compiled-path',
-                             default=from_dot_kapitan('lint', 'compiled-path', './compiled'),
-                             help='set compiled path, default is "./compiled"')
-    lint_parser.add_argument('--inventory-path',
-                             default=from_dot_kapitan('lint', 'inventory-path', './inventory'),
-                             help='set inventory path, default is "./inventory"')
-
-    init_parser = subparser.add_parser('init', help='initialize a directory with the recommended kapitan project skeleton.')
-    init_parser.add_argument('--directory',
-                             default=from_dot_kapitan('init', 'directory', '.'),
-                             help='set path, in which to generate the project skeleton, assumes directory already exists. default is "./"')
-
-    args = parser.parse_args()
-
-    logger.debug('Running with args: %s', args)
-
-    try:
-        cmd = sys.argv[1]
-    except IndexError:
-        parser.print_help()
-        sys.exit(1)
-
-    if hasattr(args, 'verbose') and args.verbose:
-        logging.basicConfig(level=logging.DEBUG,
-                            format='%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
-    elif hasattr(args, 'quiet') and args.quiet:
-        logging.basicConfig(level=logging.CRITICAL, format="%(message)s")
-    else:
         logging.basicConfig(level=logging.INFO, format="%(message)s")
+        
+        try:
+            cmd = sys.argv[1]
+        except IndexError:
+            print("Usage Options:")
+            print("kapitan -- --help")
```

Comment:
What do the "--" do?

Reply:
We must add an isolated -- argument in order to have arguments treated as Flags rather than be consumed by Python Fire. All arguments to a Fire CLI after the final standalone -- argument are treated as Flags.
Please check this https://github.com/google/python-fire/blob/master/docs/using-cli.md

Reply:
This is not a viable solution as it significantly changes the way we use the CLI

Reply:
I agree, but logically fire needs some way to distinguish between two types of flags, because ``` $ python test.py --help``` could mean you are passing argument to some function(say) fired through python-fire.